src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -458,11 +505,6 @@ private bool ReadConstructorArgumentsWithContinuation(ref ReadStack state, ref U
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void BeginRead(ref ReadStack state, ref Utf8JsonReader reader, JsonSerializerOptions options)
         {
-            if (reader.TokenType != JsonTokenType.StartObject)
```

Comment:
should this be something along the lines of `reader.TokenType != JsonTokenType.StartObject || CanHaveMetadata` now?

Reply:
`CanHaveMetadata` is always true for object converters:

https://github.com/dotnet/runtime/blob/c0db07b3f3f093b5f4cd27e1f5e8aa54adad049d/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectDefaultConverter.cs#L17